conceptnet5/builders/morphology.py

```diff
@@ -0,0 +1,57 @@
+import click
+from conceptnet5.nodes import get_uri_language, split_uri
+from conceptnet5.languages import ATOMIC_SPACE_LANGUAGES
+from conceptnet5.edges import make_edge
+from conceptnet5.uri import join_uri, Licenses
+from collections import defaultdict
+from conceptnet5.formats.msgpack_stream import MsgpackStreamWriter
+
+
+def prepare_vocab_for_morphology(language, input, output):
+    vocab_counts = defaultdict(int)
+    for line in input:
+        countstr, uri = line.strip().split(' ', 1)
+        if get_uri_language(uri) == language:
+            term = split_uri(uri)[2]
+            if language in ATOMIC_SPACE_LANGUAGES:
+                term += '_'
+            vocab_counts[term] += int(countstr)
+
+    for term, count in sorted(list(vocab_counts.items())):
+        print(count, term, file=output)
+
+
+MORPH_SOURCES = [{'process': '/s/rule/morfessor'}]
+
+
+def subwords_to_edges(language, input, output):
+    """
+    Morfessor hypothesizes ways to break words into sub-word chunks. Produce
+    edges from these sub-words that can be used in retrofitting.
+    """
+    writer = MsgpackStreamWriter(output)
+    for line in input:
+        line = line.rstrip()
+        if not line or line.startswith('#'):
+            continue
+
+        # Remove the unnecessary count ("1 ") from the start of each line
+        line = line.split(' ', 1)[1]
+        chunks = line.split(' + ')
+
+        # Strip a possible trailing underscore, which would particularly show
+        # up in the way we segment ATOMIC_SPACE_LANGUAGES (Vietnamese)
+        slug = ''.join(chunks).strip('_')
```

Comment:
Why is this variable called `slug`?

Reply:
Now that you mention it, that only makes sense in terms of my Mystery Hunt code. I'll rename it.